xlators/features/locks/src/posix.c

```diff
@@ -189,15 +194,30 @@ pl_has_xdata_requests(dict_t *xdata)
                               SLEN(GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS),
                               0};
     int i = 0;
+    uint32_t bitfield = 0;
 
     if (!xdata)
         return _gf_false;
 
-    for (i = 0; reqs[i]; i++)
-        if (dict_getn(xdata, reqs[i], reqs_size[i]))
-            return _gf_true;
+    if (dict_del_sizen(xdata, GLUSTERFS_ENTRYLK_COUNT))
+        SET_BIT(bitfield, GLUSTERFS_ENTRYLK_COUNT_BIT);
 
-    return _gf_false;
+    if (dict_del_sizen(xdata, GLUSTERFS_INODELK_COUNT))
+        SET_BIT(bitfield, GLUSTERFS_INODELK_COUNT_BIT);
+
+    if (dict_del_sizen(xdata, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS))
+        SET_BIT(bitfield, GLUSTERFS_MULTIPLE_DOM_LK_CNT_REQUESTS_BIT);
+
+    if (dict_del_sizen(xdata, GLUSTERFS_INODELK_DOM_COUNT))
+        SET_BIT(bitfield, GLUSTERFS_INODELK_DOM_COUNT_BIT);
```

Comment:
We cannot delete this one here. This key contains a string that we need to keep. In this particular case we need to do a `dict_get_sizen()`